tests/src/Interop/PInvoke/IEnumerator/IEnumeratorNative.cpp

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "IEnumeratorNative.h"
+#include <xplatform.h>
+
+extern "C" DLL_EXPORT HRESULT STDMETHODCALLTYPE GetIntegerEnumerator(int start, int count, IEnumVARIANT** ppEnum)
+{
+    if (count < 0)
+    {
+        return E_INVALIDARG;
+    }
+
+    *ppEnum = new(CoreClrAlloc(sizeof(IntegerEnumerator))) IntegerEnumerator(start, count);
+
+    return S_OK;
+}
+
+extern "C" DLL_EXPORT HRESULT STDMETHODCALLTYPE VerifyIntegerEnumerator(IEnumVARIANT* pEnum, int start, int count)
+{
+    if (count < 0)
+    {
+        return E_INVALIDARG;
+    }
+
+    HRESULT hr = S_OK;
+
+    VARIANT element;
+    ULONG numFetched;
+    
+    for(int i = start; i < start + count; ++i)
+    {
+        hr = pEnum->Next(1, &element, &numFetched);
+        if(FAILED(hr) || numFetched != 1)
+        {
+            return hr;
+        }
+
+        if (V_I4(&element) != i)
+        {
+            return E_UNEXPECTED;
+        }
+    }
```

Comment:
Please call `VariantClear()` after each loop. When copy/paste happens I want the logic to be sound.